CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.14)
```

Comment:
Hmm... There is this: https://github.com/NOAA-OWP/ngen/blob/ccce9dc1f2915b90719bff4f0f9b061367220cd2/docker/CENTOS_4.8.5_NGEN_RUN.dockerfile#L9 
... That's centos7 with GCC 4.8.5 I think, not centos4, thankfully... but I don't think we can support GCC 4.x anymore... Probably that dockerfile needs to be removed.

Reply:
Need at least GCC 5 for C++14:
https://en.cppreference.com/w/cpp/compiler_support/14

Probably a better idea to go with slightly newer, based on thoroughly explored compiler bugs in this space:
https://kokkos.github.io/kokkos-core-wiki/requirements.html
Kokkos 3.x also required C++14, and set GCC 5.3 as a minimum due to bugs in earlier versions.

Reply:
Can we leave that broader cleanup for another issue?

Reply:
Also, CentOS itself is obsolescent. RedHat sunset the unbadged rebuild a couple years back and replaced it with CentOS Stream, which is quite different from the RHEL-compatible thing. AlmaLinux took up the mantle, but even that's now at risk:
https://www.phoronix.com/news/AlmaLinux-With-RHEL-Source-Path

Reply:
Yeah we're painfully aware. Another of the dockerfiles starts with Rocky, and we use Rocky in DMOD.

Reply:
> Can we leave that broader cleanup for another issue?

Yes, at least as far as any CMake minimum version stuff is concerned, based on the comment elsewhere about NumPy needing 3.14 anyway.

Reply:
Opened #544 to do that clean-up